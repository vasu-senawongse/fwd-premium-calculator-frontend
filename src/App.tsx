import React from 'react';
import logo from './logo.png';
import './App.css';
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';
import * as R from 'ramda';

import axios from 'axios';
var DatePicker = require('reactstrap-date-picker');
interface IModel {
  name: string;
  dob: string;
  planCode: string;
  premiumPerYear: number;
  paymentFrequency: string;
  saPerYear: number;
}
interface IResponse {
  benefitTable: Array<Object>;
  deathBenefitList: Array<Object>;
  dividendBenefitList: Array<Object>;
  modalRatesList: Array<Object>;
  otherBenefitList: Array<Object>;
  quotationProductList: Array<Object>;
  survivalBenefitList: Array<Object>;
}

class App extends React.Component<
  any,
  {
    model: IModel;
    response: IResponse;
  }
> {
  constructor(props: any) {
    super(props);
    this.state = {
      model: {
        name: '',
        dob: new Date().toISOString(),
        planCode: '',
        premiumPerYear: 0,
        paymentFrequency: '',
        saPerYear: 0,
      },
      response: {
        benefitTable: [],
        deathBenefitList: [],
        dividendBenefitList: [],
        modalRatesList: [],
        otherBenefitList: [],
        quotationProductList: [],
        survivalBenefitList: [],
      },
    };
  }

  public submit() {
    axios
      .post('http://localhost:3000/get-product', this.state.model)
      .then((response) => {
        this.setState({ response: response.data });
        console.log('response: ', this.state.response);
      })
      .catch((err) => {
        console.error(err);
      });
  }

  public onValueChanged(value: any) {
    if (value == null) {
      return;
    }
    let model = this.state.model;
    model = R.merge(model, value);
    this.setState({ model });
  }

  public render() {
    const { name, dob, planCode, premiumPerYear, paymentFrequency, saPerYear } =
      this.state.model;
    return (
      <div className='App'>
        <header className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <Form>
            <FormGroup>
              <Label for='name'>Name</Label>
              <Input
                type='text'
                name='name'
                id='name'
                placeholder='Your name ...'
                value={name}
                onChange={(e) => this.onValueChanged({ name: e.target.value })}
              />{' '}
            </FormGroup>
            <FormGroup>
              <Label for='dob'>Date of Birth</Label>
              <DatePicker
                id='dob'
                name='dob'
                value={dob}
                onChange={(e: any) => this.onValueChanged({ dob: e })}
              />
            </FormGroup>
            <FormGroup>
              <Label for='planCode'>Plan</Label>
              <Input
                type='select'
                name='planCode'
                id='planCode'
                value={planCode}
                onChange={(e) =>
                  this.onValueChanged({ planCode: e.target.value })
                }
              >
                <option value=''>-- SELECT PLAN --</option>
                <option value='T11A20'>package 1 (benefit 200k)</option>
                <option value='T11A50'>package 2 (benefit 500k)</option>
                <option value='T11AM1'>package 3 (benefit 1M)</option>
              </Input>
            </FormGroup>
            <FormGroup>
              <Label for='premiumPerYear'>Premium / Year</Label>
              <Input
                type='number'
                name='premiumPerYear'
                id='premiumPerYear'
                placeholder='Premium/Year ...'
                value={premiumPerYear}
                onChange={(e) =>
                  this.onValueChanged({ premiumPerYear: e.target.value })
                }
              />
            </FormGroup>
            <FormGroup>
              <Label for='paymentFrequency'>Payment Frequency</Label>
              <Input
                type='select'
                name='paymentFrequency'
                id='paymentFrequency'
                value={paymentFrequency}
                onChange={(e) =>
                  this.onValueChanged({ paymentFrequency: e.target.value })
                }
              >
                <option value=''>-- SELECT PAYMENT FREQUENCY --</option>
                <option value='YEARLY'>YEARLY</option>
                <option value='HALFYEARLY'>HALFYEARLY</option>
                <option value='QUARTERLY'>QUARTERLY</option>
                <option value='MONTHLY'>MONTHLY</option>
              </Input>
            </FormGroup>
            <FormGroup>
              <Label for='saPerYear'>Sa / Year</Label>
              <Input
                type='number'
                name='saPerYear'
                id='saPerYear'
                placeholder='Sa/Year ...'
                value={saPerYear}
                onChange={(e) =>
                  this.onValueChanged({ saPerYear: e.target.value })
                }
              />
            </FormGroup>
            <Button color='success' onClick={() => this.submit()}>
              Submit
            </Button>
          </Form>
        </header>
      </div>
    );
  }
}

export default App;
