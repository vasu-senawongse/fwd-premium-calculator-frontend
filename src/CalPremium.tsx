import React, { useState } from 'react';
import './App.css';
import CalPremiumTable from './CalPremiumTable';
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';
import { useDispatch } from 'react-redux';
import { addTodo } from './store/calPremium/actions';
import axios from 'axios';

var DatePicker = require('reactstrap-date-picker');
const Todo = () => {
  const [model, setTodoMessage] = useState({
    name: '',
    genderCd: '',
    dob: new Date().toISOString(),
    planCode: '',
    saPerYear: 0,
    paymentFrequency: '',
    baseAnnualPremium: 0,
  });

  const dispatch = useDispatch();

  const handleButtonClick = () => {
    var todo = model;
    axios
      .post('http://localhost:5000/get-product', model)
      .then((response) => {
        const temp = response.data.quotationProductList
          ? response.data.quotationProductList[0]
          : null;
        todo.baseAnnualPremium = temp.baseAnnualPremium;
        handeOnChange({ baseAnnualPremium: temp.baseAnnualPremium });
        dispatch(addTodo(todo));
      })
      .catch((err) => {
        console.error(err);
      });
  };

  const handeOnChange = (event: any) => {
    setTodoMessage({
      ...model,
      ...event,
    });
  };

  return (
    <div>
      <Form>
        <FormGroup>
          <Label for='PM-name'>Name</Label>
          <Input
            type='text'
            name='PM-name'
            id='PM-name'
            placeholder='Your name ...'
            value={model.name}
            onChange={(e) => handeOnChange({ name: e.target.value })}
            required
          />{' '}
        </FormGroup>
        <FormGroup>
          <Label for='PM-genderCd'>Gender</Label>
          <Input
            type='select'
            name='PM-genderCd'
            id='PM-genderCd'
            value={model.genderCd}
            required
            onChange={(e) => handeOnChange({ genderCd: e.target.value })}
          >
            <option id='PM-genderCd-dafault' value=''>
              -- SELECT GENDER --
            </option>
            <option id='PM-genderCd-Male' value='MALE'>
              MALE
            </option>
            <option id='PM-genderCd-Female' value='FEMALE'>
              FEMALE
            </option>
          </Input>
        </FormGroup>
        <FormGroup>
          <Label for='PM-dob'>Date of Birth</Label>
          <DatePicker
            id='PM-dob'
            name='PM-dob'
            value={model.dob}
            dateFormat='DD/MM/YYYY'
            required
            onChange={(e: any) => handeOnChange({ dob: e })}
          />
        </FormGroup>
        <FormGroup>
          <Label for='PM-planCode'>Plan</Label>
          <Input
            type='select'
            name='PM-planCode'
            id='PM-planCode'
            value={model.planCode}
            required
            onChange={(e) => handeOnChange({ planCode: e.target.value })}
          >
            <option value=''>-- SELECT PLAN --</option>
            <option id='PM-T11A20' value='T11A20'>
              package 1 (benefit 200k)
            </option>
            <option id='PM-T11A50' value='T11A50'>
              package 2 (benefit 500k)
            </option>
            <option id='PM-T11AM1' value='T11AM1'>
              package 3 (benefit 1M)
            </option>
          </Input>
        </FormGroup>
        <FormGroup>
          <Label for='PM-saPerYear'>Sa / Year</Label>
          <Input
            type='number'
            name='PM-saPerYear'
            id='PM-saPerYear'
            placeholder='Sum Assured/Year ...'
            value={model.saPerYear}
            required
            onChange={(e) => handeOnChange({ saPerYear: e.target.value })}
          />
        </FormGroup>
        <FormGroup>
          <Label for='PM-paymentFrequency'>Payment Frequency</Label>
          <Input
            type='select'
            name='PM-paymentFrequency'
            id='PM-paymentFrequency'
            value={model.paymentFrequency}
            required
            onChange={(e) =>
              handeOnChange({ paymentFrequency: e.target.value })
            }
          >
            <option id='PM-paymentFrequency-default' value=''>
              -- SELECT PAYMENT FREQUENCY --
            </option>
            <option id='PM-paymentFrequency-YEARLY' value='YEARLY'>
              YEARLY
            </option>
            <option id='PM-paymentFrequency-HALFYEARLY' value='HALFYEARLY'>
              HALFYEARLY
            </option>
            <option id='PM-paymentFrequency-QUARTERLY' value='QUARTERLY'>
              QUARTERLY
            </option>
            <option id='PM-paymentFrequency-MONTHLY' value='MONTHLY'>
              MONTHLY
            </option>
          </Input>
        </FormGroup>

        <Button
          id='PM-submit'
          color='success'
          disabled={
            model.name == '' ||
            model.genderCd == '' ||
            model.dob == null ||
            model.saPerYear == 0 ||
            model.planCode == '' ||
            model.paymentFrequency == ''
          }
          onClick={() => handleButtonClick()}
        >
          Submit
        </Button>

        <FormGroup>
          <Label for='PM-baseAnnualPremium'>Annual Premium</Label>
          <Input
            type='text'
            id='PM-baseAnnualPremium'
            name='PM-baseAnnualPremium'
            disabled={true}
            value={model.baseAnnualPremium}
          />{' '}
        </FormGroup>
      </Form>
      <CalPremiumTable />
    </div>
  );
};

export default Todo;
