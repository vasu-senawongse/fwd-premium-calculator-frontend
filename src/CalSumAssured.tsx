import React, { useState } from 'react';
import './App.css';
import CalSumAssuredTable from './CalSumAssuredTable';
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';
import { useDispatch } from 'react-redux';
import { addTodo } from './store/calSumAssured/actions';
import axios from 'axios';

var DatePicker = require('reactstrap-date-picker');
const Todo = () => {
  const [model, setTodoMessage] = useState({
    name: '',
    genderCd: '',
    dob: new Date().toISOString(),
    planCode: '',
    premiumPerYear: 0,
    paymentFrequency: '',
    baseSumAssured: 0,
  });

  const dispatch = useDispatch();

  const handleButtonClick = () => {
    var todo = model;
    axios
      .post('http://localhost:5000/get-product', model)
      .then((response) => {
        const temp = response.data.quotationProductList
          ? response.data.quotationProductList[0]
          : null;
        todo.baseSumAssured = temp.baseSumAssured;
        handeOnChange({ baseSumAssured: temp.baseSumAssured });
        dispatch(addTodo(todo));
      })
      .catch((err) => {
        console.error(err);
      });
  };

  const handeOnChange = (event: any) => {
    setTodoMessage({
      ...model,
      ...event,
    });
  };

  return (
    <div>
      <Form>
        <FormGroup>
          <Label for='SA-name'>Name</Label>
          <Input
            type='text'
            name='SA-name'
            id='SA-name'
            placeholder='Your name ...'
            value={model.name}
            required
            onChange={(e) => handeOnChange({ name: e.target.value })}
          />{' '}
        </FormGroup>
        <FormGroup>
          <Label for='SA-genderCd'>Gender</Label>
          <Input
            type='select'
            name='SA-genderCd'
            id='SA-genderCd'
            required
            value={model.genderCd}
            onChange={(e) => handeOnChange({ genderCd: e.target.value })}
          >
            <option value=''>-- SELECT GENDER --</option>
            <option id='SA-genderCd-Male' value='MALE'>
              MALE
            </option>
            <option id='SA-genderCd-Female' value='FEMALE'>
              FEMALE
            </option>
          </Input>
        </FormGroup>
        <FormGroup>
          <Label for='dob'>Date of Birth</Label>
          <DatePicker
            id='SA-dob'
            name='SA-dob'
            value={model.dob}
            required
            dateFormat='DD/MM/YYYY'
            onChange={(e: any) => handeOnChange({ dob: e })}
          />
        </FormGroup>
        <FormGroup>
          <Label for='SA-planCode'>Plan</Label>
          <Input
            type='select'
            name='SA-planCode'
            id='SA-planCode'
            required
            value={model.planCode}
            onChange={(e) => handeOnChange({ planCode: e.target.value })}
          >
            <option value=''>-- SELECT PLAN --</option>
            <option id='SA-T11A20' value='T11A20'>
              package 1 (benefit 200k)
            </option>
            <option id='SA-T11A50' value='T11A50'>
              package 2 (benefit 500k)
            </option>
            <option id='SA-T11AM1' value='T11AM1'>
              package 3 (benefit 1M)
            </option>
          </Input>
        </FormGroup>
        <FormGroup>
          <Label for='SA-premiumPerYear'>Premium / Year</Label>
          <Input
            type='number'
            name='SA-premiumPerYear'
            id='SA-premiumPerYear'
            required
            placeholder='Premium/Year ...'
            value={model.premiumPerYear}
            onChange={(e) => handeOnChange({ premiumPerYear: e.target.value })}
          />
        </FormGroup>
        <FormGroup>
          <Label for='SA-paymentFrequency'>Payment Frequency</Label>
          <Input
            type='select'
            name='SA-paymentFrequency'
            id='SA-paymentFrequency'
            value={model.paymentFrequency}
            required
            onChange={(e) =>
              handeOnChange({ paymentFrequency: e.target.value })
            }
          >
            <option value=''>-- SELECT PAYMENT FREQUENCY --</option>
            <option id='SA-paymentFrequency-YEARLY' value='YEARLY'>
              YEARLY
            </option>
            <option id='SA-paymentFrequency-HALFYEARLY' value='HALFYEARLY'>
              HALFYEARLY
            </option>
            <option id='SA-paymentFrequency-QUARTERLY' value='QUARTERLY'>
              QUARTERLY
            </option>
            <option id='SA-paymentFrequency-MONTHLY' value='MONTHLY'>
              MONTHLY
            </option>
          </Input>
        </FormGroup>

        <Button
          id='SA-submit'
          color='success'
          disabled={
            model.name == '' ||
            model.genderCd == '' ||
            model.dob == null ||
            model.premiumPerYear == 0 ||
            model.planCode == '' ||
            model.paymentFrequency == ''
          }
          onClick={() => handleButtonClick()}
        >
          Submit
        </Button>

        <FormGroup>
          <Label for='SA-baseSumAssured'>Sum Assured</Label>
          <Input
            id='SA-baseSumAssured'
            name='SA-baseSumAssured'
            type='text'
            disabled={true}
            value={model.baseSumAssured}
          />{' '}
        </FormGroup>
      </Form>
      <CalSumAssuredTable />
    </div>
  );
};

export default Todo;
