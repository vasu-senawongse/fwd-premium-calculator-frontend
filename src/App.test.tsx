import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import CalSumAssured from './CalSumAssured';
import CalPremium from './CalPremium';
import { Provider } from 'react-redux';
import configureStore from './store';
describe('App', () => {
  let store: any;
  beforeEach(() => {
    store = configureStore();
  });

  test('Should renders the Calc SA', () => {
    const component = shallow(<App />);
    expect(component.contains(<CalSumAssured />)).toBe(true);
  });

  test('Should renders the Calc Premium', () => {
    const component = shallow(<App />);
    expect(component.contains(<CalPremium />)).toBe(true);
  });

  test('Should renders Calc SA Tab', () => {
    const component = shallow(<App />);
    expect(component.find('#tab1').exists()).toBe(true);
  });

  test('Should renders Calc Premium Tab', () => {
    const component = shallow(<App />);
    expect(component.find('#tab2').exists()).toBe(true);
  });
});
