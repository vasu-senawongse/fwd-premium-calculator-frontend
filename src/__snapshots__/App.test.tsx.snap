// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.png"
    />
    <Nav
      tabs={true}
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <NavLink
          className="active"
          onClick={[Function]}
          tag="a"
        >
          Calc SA
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          className=""
          onClick={[Function]}
          tag="a"
        >
          Calc Premium
        </NavLink>
      </NavItem>
    </Nav>
    <TabContent
      activeTab="1"
      tag="div"
    >
      <TabPane
        tabId="1"
        tag="div"
      >
        <Todo />
      </TabPane>
      <TabPane
        tabId="2"
        tag="div"
      >
        <Todo />
      </TabPane>
    </TabContent>
  </header>
</div>
`;
